server/service/transport_error.go

```diff
@@ -119,6 +119,10 @@ func encodeError(ctx context.Context, err error, w http.ResponseWriter) {
 			w.WriteHeader(http.StatusUnauthorized)
 			errMap["node_invalid"] = true
 		} else {
+			// TODO: osqueryError is not always the result of an internal error on
+			// our side, it is also used to represent a client error (invalid data,
+			// e.g. malformed json, carve too large, etc., so 4xx), are we returning
+			// a 500 because of some osquery-specific requirement?
```

Comment:
Should be part of a larger-scale review of http status codes in our API: https://github.com/fleetdm/fleet/issues/4406.